### PR TITLE
fix: address time interval overrride on comparison of metrics

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: N/A

### Description:

After this merge: https://github.com/lightdash/lightdash/pull/12748, I found some issues when interacting with the Date zoom + metric comparison. This PR addresses that.

This PR ensures that Date zoom works ok when comparing metric A with another metric. 
Granularity is respected. 

Values are parsed so that the y axis range is correct - can revert this change if we want to tackle this differently.

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/services/MetricsExplorerService/MetricsExplorerService.ts [hunk 1]
```diff
@@ -202,8 +203,8 @@ export class MetricsExplorerService<
                         );
                     }
 
-                    const compareMetricDimensionGrain = dateRange
-                        ? getGrainForDateRange(dateRange)
+                    const compareMetricDimensionGrain = timeDimensionOverride
```
Comment (joaoviana) [inline packages/backend/src/services/MetricsExplorerService/MetricsExplorerService.ts:206]: we should keep the time dimension or the previous interval to get correct metric comparison

File: packages/frontend/src/features/metricsCatalog/components/MetricPeekModal.tsx [hunk 2]
```diff
@@ -80,21 +80,51 @@ export const MetricPeekModal: FC<Props> = ({ opened, onClose }) => {
...
-            setSelectedMetric(metric);
+            setSelectedMetric(metric ?? null);
         },
         [
             metricsWithTimeDimensionsQuery.data,
             metricsWithTimeDimensionsQuery.isSuccess,
         ],
     );
 
+    const handleComparisonTypeChange = useCallback(
```
Comment (joaoviana) [inline packages/frontend/src/features/metricsCatalog/components/MetricPeekModal.tsx:103]: when going from one comparison mode to another, clear the range, metric, etc

File: packages/frontend/src/features/metricsCatalog/hooks/useRunMetricExplorerQuery.ts [hunk 3]
```diff
@@ -84,7 +85,12 @@ export const useRunMetricExplorerQuery = ({
             !!exploreName &&
             !!metricName &&
             !!comparison &&
-            !!dateRange,
+            !!dateRange &&
+            (comparison.type !== MetricExplorerComparison.DIFFERENT_METRIC ||
+                (comparison.type ===
+                    MetricExplorerComparison.DIFFERENT_METRIC &&
```
Comment (joaoviana) [inline packages/frontend/src/features/metricsCatalog/hooks/useRunMetricExplorerQuery.ts:91]: only enable the api request for different metric if all required params are set

File: packages/backend/src/services/CatalogService/CatalogService.ts [hunk 4]
```diff
@@ -975,7 +977,9 @@ export class CatalogService<
 
             timeDimension = {
                 field: firstAvailableTimeDimension.name,
-                interval: DEFAULT_METRICS_EXPLORER_TIME_INTERVAL,
+                interval:
+                    timeIntervalOverride ??
+                    DEFAULT_METRICS_EXPLORER_TIME_INTERVAL,
```
[the remark this answers is not in the document]
Comment (IrakliJani) [reply packages/backend/src/services/CatalogService/CatalogService.ts:982]: good catch

[the remark this answers is not in the document]
Comment (IrakliJani) [reply packages/backend/src/services/CatalogService/CatalogService.ts:982]: must be the bad merge/rebase

File: packages/backend/src/services/MetricsExplorerService/MetricsExplorerService.ts (hunk 1 above)
Comment (IrakliJani) [reply packages/backend/src/services/MetricsExplorerService/MetricsExplorerService.ts:206]: correct. its done in the UI now so there is no need for that call

File: packages/backend/src/services/MetricsExplorerService/MetricsExplorerService.ts [hunk 5]
```diff
@@ -262,6 +263,8 @@ export class MetricsExplorerService<
             }
         }
 
+        console.log({ compareMetric });
```
Comment (IrakliJani) [inline packages/backend/src/services/MetricsExplorerService/MetricsExplorerService.ts]: log

File: packages/frontend/src/features/metricsCatalog/components/MetricPeekModal.tsx [hunk 6]
```diff
@@ -199,11 +227,22 @@ export const MetricPeekModal: FC<Props> = ({ opened, onClose }) => {
...
+                setTimeDimensionOverride({
+                    ...timeDimensionBaseField,
+                    interval: timeInterval,
+                });
+            }
+        },
+        [timeDimensionBaseField],
+    );
 
+    console.log({ timeDimensionOverride });
```
Comment (IrakliJani) [inline packages/frontend/src/features/metricsCatalog/components/MetricPeekModal.tsx:245]: log

File: packages/frontend/src/features/metricsCatalog/hooks/useRunMetricExplorerQuery.ts (hunk 3 above)
Comment (IrakliJani) [reply packages/frontend/src/features/metricsCatalog/hooks/useRunMetricExplorerQuery.ts:91]: my suggestion here is to have query params passed to this hook and then use `enabled` prop to control it from the UI component

**Review by IrakliJani — COMMENTED** [review]
approving with suggestions

File: packages/backend/src/services/MetricsExplorerService/MetricsExplorerService.ts (hunk 5 above)
Comment (joaoviana) [reply packages/backend/src/services/MetricsExplorerService/MetricsExplorerService.ts]: done ty!

File: packages/frontend/src/features/metricsCatalog/hooks/useRunMetricExplorerQuery.ts (hunk 3 above)
Comment (joaoviana) [reply packages/frontend/src/features/metricsCatalog/hooks/useRunMetricExplorerQuery.ts]: thoughts on this? https://github.com/lightdash/lightdash/pull/12765/commits/b4ec874420b368ee8ccac435bcbf446562e5aefc

**Review by IrakliJani — APPROVED** [review]

**Pull request merged**